### PR TITLE
HKG: Car Port for Kia Carnival 2023 (HDA1)

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with stock system: `409c9409979a8abc|2023-03-25--11-30-08`
- [x] route with openpilot:
  - U.S. Model: `de59124955b921d8|2023-06-24--00-12-50`
  - Chinese Model: `409c9409979a8abc|2023-04-27--00-13-04`
- [x] harness type:
  - U.S. Model: [Hyundai A](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)
  - Chinese Model: [Hyundai K](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)
- [x] architecture: CAN-FD

**Physical setup**
- Top port behind comma three
  - comma Power <-> First Harness Box <-> comma three USB C/Right-Angle C <-> comma three
- Bottom port behind comma three
  - Camera Harness <-> Second Harness Box <-> USB C/C 3.1 <-> Red Panda <-> USB A/A <-> USB A to C **OTG Adapter** <-> comma three
- Link to all physical parts during install: 
  - [comma three](https://comma.ai/shop/comma-three)
  - [car harness kit](https://comma.ai/shop/car-harness)
  - [CAN FD panda kit](https://comma.ai/shop/can-fd-panda-kit)

Thanks to the community Kia Carnival 2023 owners cbmhighflier (U.S.) and `RTK#4465` (Discord) (China only).

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -235,6 +236,9 @@ def init_make(self, CP: car.CarParams):
     HyundaiCarInfo("Kia EV6 (without HDA II) 2022", "Highway Driving Assist", harness=Harness.hyundai_l),
     HyundaiCarInfo("Kia EV6 (with HDA II) 2022", "Highway Driving Assist II", harness=Harness.hyundai_p)
   ],
+  CAR.KIA_CARNIVAL_4TH_GEN: [
+    HyundaiCarInfo("Kia Carnival (China) 2023", harness=Harness.hyundai_k),
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:240]: Why China only?

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py:240]: > Why China only?

The user mentioned that US and China models use different harnesses, and we currently don't have any confirmation from any users with the US model yet. We can always change it once we have confirmation.

Comment (sshane) [reply selfdrive/car/hyundai/values.py:240]: Could it be the same exception as what we have for the EV6 and Ioniq 5? "Southeast Asia only"?

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py]: We have a U.S. version that is confirmed to use a different harness. Will have them separated in this PR.

**Comment by sshane** [comment]
Discussed on Discord: we should have a test for new Hyundai CAN FD ports that asserts that the message we override and send to disable LFA exists, would avoid cases where we merge a port where it doesn't work and LFA is overriding lane changes and whatnot.

**Comment by sshane** [comment]
@sunnyhaibin going to merge now without the chinese FW since those rlogs have expired unfortunately. looks like the user is active so I request upload of a recent drive, let me know if you reach out to him and get permission to use the new route (any other is fine too)

File: RELEASES.md [hunk 2]
```diff
@@ -3,13 +3,14 @@ Version 0.9.4 (2023-XX-XX)
 * Navigate on openpilot
   * When navigation has a destination openpilot will input the map information into the model, generally improving behavior
   * When navigating on openpilot, openpilot will keep left or right appropriately at forks/exits and take turns
-  * When navigating on openpilot, lane change behavior is unchanged and still activated by the driver 
+  * When navigating on openpilot, lane change behavior is unchanged and still activated by the driver
```
Comment (sshane) [inline RELEASES.md:6]:
```suggestion
  * When navigating on openpilot, lane change behavior is unchanged and still activated by the driver 
```

File: RELEASES.md [hunk 3]
```diff
@@ -3,13 +3,14 @@ Version 0.9.4 (2023-XX-XX)
...
 * UI updates
   * Navigation settings moved to home screen and map
 * UI alerts rework
   * Border color always shows engagement status. Blue means disengaged, green means engaged, and grey means engaged with human overriding
-  * Alerts are shown inside the border. Black/grey means info, orange means warning, and red means critical alert 
+  * Alerts are shown inside the border. Black/grey means info, orange means warning, and red means critical alert
```
Comment (sshane) [inline RELEASES.md:11]:
```suggestion
  * Alerts are shown inside the border. Black/grey means info, orange means warning, and red means critical alert 
```

File: RELEASES.md (hunk 2 above)
Comment (sshane) [inline RELEASES.md:6]:
```suggestion
  * When navigating on openpilot, lane change behavior is unchanged and still activated by the driver 
```

File: RELEASES.md (hunk 3 above)
Comment (sshane) [inline RELEASES.md:11]:
```suggestion
  * Alerts are shown inside the border. Black/grey means info, orange means warning, and red means critical alert 
```

File: selfdrive/car/tests/routes.py [hunk 4]
```diff
@@ -98,6 +98,7 @@
   CarTestRoute("bf43d9df2b660eb0|2021-09-23--14-16-37", HYUNDAI.SANTA_FE_2022),
   CarTestRoute("37398f32561a23ad|2021-11-18--00-11-35", HYUNDAI.SANTA_FE_HEV_2022),
   CarTestRoute("656ac0d830792fcc|2021-12-28--14-45-56", HYUNDAI.SANTA_FE_PHEV_2022, segment=1),
+  CarTestRoute("de59124955b921d8|2023-06-24--00-12-50", HYUNDAI.KIA_CARNIVAL_4TH_GEN),  # U.S. model
```
Comment (sshane) [inline selfdrive/car/tests/routes.py:101]:
```suggestion
  CarTestRoute("de59124955b921d8|2023-06-24--00-12-50", HYUNDAI.KIA_CARNIVAL_4TH_GEN),
```

**Comment by DDavidL** [comment]
> 

I'm glad to provide Chinese FW rlogs. How can I share them with you? Or is it okay if I just upload some recent routes with rlogs from my C3

**Comment by sunnyhaibin** [comment]
PR opened to support the Chinese model:
- https://github.com/commaai/openpilot/pull/28918

**Pull request merged**